File: leetCode/easy/2114-maxNumberOfWordsInSentences.ts
import { assertEquals } from "https://deno.land/std@0.182.0/testing/asserts.ts";
import { describe, it } from "https://deno.land/std@0.182.0/testing/bdd.ts";

function maxNumberOfWordsInSentences(sentences: string[]) {
  return Math.max(...sentences.map((sentence) => sentence.split(" ").length));
}

export { maxNumberOfWordsInSentences };

describe("maxNumberOfWordsInSentences", () => {
  it("1. should return the max number of words in a sentence", () => {
    const sentences = ["i love you", "island", "iroman", "i love leetcode"];
    assertEquals(maxNumberOfWordsInSentences(sentences), 3);
  });

  it("2. should return the max number of words in a sentence", () => {
    const sentences = ["hello world", "leetcode"];
    assertEquals(maxNumberOfWordsInSentences(sentences), 2);
  });

  it("3. should return the max number of words in a sentence", () => {
    const sentences = [
      "alice and bob love leetcode",
      "i think so too",
      "this is great thanks very much",
    ];
    assertEquals(maxNumberOfWordsInSentences(sentences), 6);
  });
});

File: leetCode/easy/2114-maxNumberOfWordsInSentences.jl
function maxNumberOfWordsInSentences(sentences::Vector{String})
  return map(sentence -> split(sentence, " ") |> length, sentences) |> maximum
end

export maxNumberOfWordsInSentences

using Test

Test.@testset "1. maxNumberOfWordsInSentences" begin
  sentences = ["Hello World", "I am a leetcode"]
  result = 4
  Test.@test maxNumberOfWordsInSentences(sentences) == result
end

Test.@testset "2. maxNumberOfWordsInSentences" begin
  sentences = ["alice and bob love leetcode", "i think so too", "this is great thanks very much"]
  result = 6
  Test.@test maxNumberOfWordsInSentences(sentences) == result
end

Test.@testset "3. maxNumberOfWordsInSentences" begin
  sentences = ["please wait", "continue to fight", "continue to win"]
  result = 3
  Test.@test maxNumberOfWordsInSentences(sentences) == result
end